### ISSUE TITLE
Decrease distribution package size

### ISSUE THREAD

**Issue by mysterymath** [llvm-mos/llvm-mos#71]
Something is wonky with the sizes of our packages; the Arch Linux packages for LLVM and Clang together are under 80MB, while our distribution is at 535MB. It's possible that the default compression used by Arch packages is tremendously better than ZIP, but that's unlikely to make that big of a difference unless we're completely miscompressing the ZIPs.

https://archlinux.org/packages/extra/x86_64/llvm/
https://archlinux.org/packages/extra/x86_64/clang/

**Comment by mysterymath** [comment]
The discrepancy appears to be due to the use of dynamic linking in distribution binaries. Both our distribution and the LLVM standalone distribution (their Github Releases) use static linking, so each binary contains a full copy of LLVM. However, for us to use dynamic linking, we'd need to namespace the shared libraries to separate them from the default ones (i.e., we'd link against libclang-mos, not libclang), and we'd need to install these shared libraries into the appropriate directories so the loader can pick them up. On Windows the same directory as the executables suffices, but on POSIX these generally need to be in system paths.

Renaming and removing from v0.1.0 milestone, as this would need to happen after we have installers.

**Comment by johnwbyrd** [comment]
The LLVM docs suggest using LLVM_BUILD_LLVM_DYLIB and LLVM_LINK_LLVM_DYLIB on non-Windows platforms.  I think we should come back to this issue once we are feature complete.

**Comment by mysterymath** [comment]
Yeah, we can't take any action until we have installables on POSIX.

**Comment by mysterymath** [comment]
We actually can deal with this using rpath or equivalent on Linux and macOS and by placing the dlls into the binary directory on Windows. Learned how to do this when creating a hermetic LLDB distribution at my dayjob; it's not fun, but it works.
